keps/973-workload-priority/README.md

```diff
@@ -0,0 +1,484 @@
+
+```yaml
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: pi
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: sample-priority
+spec:
+  slotsPerWorker: 1
+  runPolicy:
+    cleanPodPolicy: Running
+    ttlSecondsAfterFinished: 60
+  sshAuthMountPath: /home/mpiuser/.ssh
+  mpiReplicaSpecs:
+    Launcher:
+      replicas: 1
+      template:
+        spec:
+          containers:
+          - image: mpioperator/mpi-pi:openmpi
+            name: mpi-launcher
+            securityContext:
+              runAsUser: 1000
+            command:
+            - mpirun
+            args:
+            - -n
+            - "2"
+            - /home/mpiuser/pi
+            resources:
+              limits:
+                cpu: 1
+                memory: 1Gi
+```
+
+#### 6. A jobFramework specifies only `priorityClass`
+
+When creating this yaml, the `PriorityClass` high-priority is used for the `workload's priority`.
+This is basically same as current implementation of workload.
+
+```yaml
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: pi
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  slotsPerWorker: 1
+  runPolicy:
+    cleanPodPolicy: Running
+    ttlSecondsAfterFinished: 60
+    schedulingPolicy:
+      priorityClass: high-priority
+  sshAuthMountPath: /home/mpiuser/.ssh
+  mpiReplicaSpecs:
+    Launcher:
+      replicas: 1
+      template:
+        spec:
+          containers:
+          - image: mpioperator/mpi-pi:openmpi
+            name: mpi-launcher
+            securityContext:
+              runAsUser: 1000
+            command:
+            - mpirun
+            args:
+            - -n
+            - "2"
+            - /home/mpiuser/pi
+            resources:
+              limits:
+                cpu: 1
+                memory: 1Gi
+```
+
+### How to expand Priority utility
+
+When referencing the `priorityClass` in the [Priority utility](https://github.com/kubernetes-sigs/kueue/blob/ba404ad282c35cf1d6b15d07643935fffbcc1835/pkg/util/priority/priority.go) function on workload, first we check the `PriorityClassSource` field.
+If `PriorityClassSource` has the value `kueue.x-k8s.io/workloadpriorityclass`, `workloadPriorityClass` is searched.
+If `PriorityClassSource` has the value `scheduling.k8s.io/priorityclass`, `priorityClass` is searched.
+
+### Where workload's Priority is used
+
+The priority of workloads is utilized in queuing, preemption, and other scheduling processes in Kueue.
+With the introduction of `workloadPriorityClass`, there is no change in the places where priority is used in Kueue.
+It just enables the usage of `workloadPriorityClass` as the priority.
+
+### Workload's priority values are always mutable
+
+`Priority`, `PriorityClassName` and `PriorityClassSource` fields is part of the workload spec and always mutable because it might be useful for the preemption.  
```

Comment:
nit

```suggestion
`Priority`, `PriorityClassName`, and `PriorityClassSource` fields are part of the workload spec and always mutable because it might be useful for the preemption.  
```

Reply:
Thanks! Fixed